### PR TITLE
Precise nodejs version, how to switch and node-gyp dependency on Windows

### LINKED ISSUE #663: Documentation for compiling assets is incomplete
The page https://devdocs.prestashop.com/1.7/development/compile-assets/ does not state prerequisites for compiling the assets.

Using the latest stable release `1.7.6.7` on Windows 10.

To be able to run `npm install` in `themes/classic/_dev/`, there are the following prerequisites:
- use `node 10.22.0`. The version of `node-sass` used is not compatible with node 12.x (see https://github.com/sass/node-sass/issues/2794).
- install `windows-build-tools` using `npm i --global windows-build-tools`. This is required for `node-gyp`to work (see https://github.com/nodejs/node-gyp#on-windows)

I am still not able to get to work `npm run build` in the admin themes directories. There are several errors.

See also https://stackoverflow.com/questions/63454346

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Nodejs needed some precisions for windows, and to explain why some version would need an older nodejs version
| Fixed ticket? | Fixes https://github.com/PrestaShop/docs/issues/663

### CODE REVIEW

File: src/content/1.7/development/compile-assets.md [hunk 1]
```diff
@@ -9,7 +9,11 @@ Some components in PrestaShop, like Javascript or SCSS files, need to be compile
 
 ## Requirements
 
-We use [Webpack](https://webpack.js.org/) to compile assets. You only need NodeJS from 8.x to 12.x ([get it here](https://nodejs.org/)), NPM will take care of it all.
+PrestaShop development is following tools version, such as webpack which could be incompatible with some NodeJS versions. This means that at the time we developed one version of PrestaShop, we may used an older version of NodeJS. We strongly recommend you ton install a package such as ['n'](https://www.npmjs.com/package/n?activeTab=versions) or [nvm](https://github.com/nvm-sh/nvm) to switch between versions.
```
Comment (atomiix) [inline src/content/1.7/development/compile-assets.md:12]:
```suggestion
PrestaShop development is following tools version, such as webpack which could be incompatible with some NodeJS versions. This means that at the time we developed one version of PrestaShop, we may used an older version of NodeJS. We strongly recommend you to install a package such as ['n'](https://www.npmjs.com/package/n?activeTab=versions) or [nvm](https://github.com/nvm-sh/nvm) to switch between versions.
```

File: src/content/1.7/development/compile-assets.md [hunk 2]
```diff
@@ -9,7 +9,11 @@ Some components in PrestaShop, like Javascript or SCSS files, need to be compile
 
 ## Requirements
 
-We use [Webpack](https://webpack.js.org/) to compile assets. You only need NodeJS from 8.x to 12.x ([get it here](https://nodejs.org/)), NPM will take care of it all.
+PrestaShop development is following tools version, such as webpack which could be incompatible with some NodeJS versions. This means that at the time we developed one version of PrestaShop, we may used an older version of NodeJS. We strongly recommend you to install a package such as ['n'](https://www.npmjs.com/package/n?activeTab=versions) or [nvm](https://github.com/nvm-sh/nvm) to switch between versions.
+
+We use [Webpack](https://webpack.js.org/) to compile assets. Currently, you only need NodeJS 12.x ([get it here](https://nodejs.org/)), NPM will take care of it all.
```
Comment (eternoendless) [inline src/content/1.7/development/compile-assets.md:14]: Avoid using relative time-bound words like "Currently" or "recently" on documentation. Some people might need to compile assets in previous versions (unlikely, but who knows). You need to keep instructions on how to compile previous versions.

File: src/content/1.7/development/compile-assets.md [hunk 3]
```diff
@@ -9,7 +9,11 @@ Some components in PrestaShop, like Javascript or SCSS files, need to be compile
 
 ## Requirements
 
-We use [Webpack](https://webpack.js.org/) to compile assets. You only need NodeJS from 8.x to 12.x ([get it here](https://nodejs.org/)), NPM will take care of it all.
+PrestaShop development is following tools version, such as webpack which could be incompatible with some NodeJS versions. This means that at the time we developed one version of PrestaShop, we may used an older version of NodeJS. We strongly recommend you to install a package such as ['n'](https://www.npmjs.com/package/n?activeTab=versions) or [nvm](https://github.com/nvm-sh/nvm) to switch between versions.
```
Comment (eternoendless) [inline src/content/1.7/development/compile-assets.md:12]: > PrestaShop development is following tools version, such as webpack which could be incompatible with some NodeJS versions. This means that at the time we developed one version of PrestaShop, we may used an older version of NodeJS.

I think this part could benefit from som rewording for clarity

File: src/content/1.7/development/compile-assets.md [hunk 4]
```diff
@@ -9,7 +9,11 @@ Some components in PrestaShop, like Javascript or SCSS files, need to be compile
...
+
+We use [Webpack](https://webpack.js.org/) to compile assets. Currently, you only need NodeJS 12.x ([get it here](https://nodejs.org/)), NPM will take care of it all.
+
+On Windows, you'll probably need to install `windows-build-tools` using `npm i --global windows-build-tools` in order to make [node-gyp](https://github.com/nodejs/node-gyp#on-windows) working.
```
Comment (eternoendless) [inline src/content/1.7/development/compile-assets.md:16]:
````suggestion
### Windows

Windows users might need to install [windows-build-tools](https://www.npmjs.com/package/@xpack-dev-tools/windows-build-tools) in order to make [node-gyp](https://github.com/nodejs/node-gyp#on-windows) work.
````

**Comment by matks** [comment]
Hello,

I have modified your PR to target the `1.7.x` branch.

We have modified the devdocs inner working in order to welcome multiple documentation versions. This is because soon we will have a 1.7 documentation and a 8.0 documentation, matching PrestaShop 1.7 or PrestaShop 8.0 softwares.

Read more on https://devdocs.prestashop.com/1.7/contribute/documentation/how/documentation-versions/

Please also note that now that we have multiple versions inside one documentation, we had to build a workflow that can handle this setup. This repository https://github.com/PrestaShop/docs now only contains the content of the documentation, and https://github.com/PrestaShop/devdocs-site contains the sources and the deployment workflow.

Because of this, some files have moved and you might have to rebase your Pull Request to match the new structure of the repository.

File: development/compile-assets.md [hunk 5]
```diff
@@ -9,7 +9,20 @@ Some components in PrestaShop, like Javascript or SCSS files, need to be compile
 
 ## Requirements
 
-We use [Webpack](https://webpack.js.org/) to compile assets. You only need NodeJS from 8.x to 12.x ([get it here](https://nodejs.org/)), NPM will take care of it all.
+We use [Webpack](https://webpack.js.org/) to compile assets. PrestaShop is using NodeJS ([get it here](https://nodejs.org/)), NPM will take care of it all.
```
Comment (eternoendless) [inline development/compile-assets.md]:
```suggestion
PrestaShop uses [Webpack](https://webpack.js.org/) to compile assets, which requires NodeJS ([get it here](https://nodejs.org/)).
```

File: development/compile-assets.md [hunk 6]
```diff
@@ -9,7 +9,20 @@ Some components in PrestaShop, like Javascript or SCSS files, need to be compile
...
+
+Here is a list of compatibility:
```
Comment (eternoendless) [inline development/compile-assets.md]: You will need a different NodeJS version depending on the PrestaShop version you're working with:

File: development/compile-assets.md [hunk 7]
```diff
@@ -9,7 +9,20 @@ Some components in PrestaShop, like Javascript or SCSS files, need to be compile
...
+
+| PrestaShop Versions | NodeJS Versions |
+|---------------------|-----------------|
+| under 1.7.7         | 8.x and older   |
```
Comment (eternoendless) [inline development/compile-assets.md]:
```suggestion
| 1.7.0~1.7.6          | 8.x   |
```

File: development/compile-assets.md [hunk 8]
```diff
@@ -9,7 +9,20 @@ Some components in PrestaShop, like Javascript or SCSS files, need to be compile
...
+| 1.7.7               | 10.x            |
+| 1.7.8               | 12.x and 14.x   |
```
Comment (eternoendless) [inline development/compile-assets.md]: We are still using node 10 for sanity tests (see https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/.github/workflows/sanity-74.yml#L39). Is it something that we should fix?

Also, what is the recommended version?

File: development/compile-assets.md [hunk 9]
```diff
@@ -9,7 +9,20 @@ Some components in PrestaShop, like Javascript or SCSS files, need to be compile
...
+
+
+If you can't manage to compile assets, you may be using an older version of PrestaShop. We strongly recommend you ton install a package such as ['n'](https://www.npmjs.com/package/n?activeTab=versions) or [nvm](https://github.com/nvm-sh/nvm) to try with an older version of NodeJS.
```
Comment (eternoendless) [inline development/compile-assets.md]: I suppose you meant this:
```suggestion
If you are struggling to compile assets, you might be using the wrong NodeJs version. We strongly recommend you to use a version manager such as ['n'](https://www.npmjs.com/package/n) or [nvm](https://github.com/nvm-sh/nvm) so you can easily switch between NodeJS versions.
```

File: development/compile-assets.md [hunk 10]
```diff
@@ -9,7 +9,20 @@ Some components in PrestaShop, like Javascript or SCSS files, need to be compile
...
+
+On Windows, you'll probably need to install `windows-build-tools` using `npm i --global windows-build-tools` in order to make [node-gyp](https://github.com/nodejs/node-gyp#on-windows) working.
```
Comment (eternoendless) [inline development/compile-assets.md]:
```suggestion
On Windows, you'll probably need to install `windows-build-tools` using `npm i --global windows-build-tools` in order to [make node-gyp work](https://github.com/nodejs/node-gyp#on-windows).
```

File: development/compile-assets.md (hunk 8 above)
Comment (NeOMakinG) [reply development/compile-assets.md]: As 1.7.8.x is compatible with latest LTS, I think we should make tests compatibles with the latest LTS too

We should use the same version everywhere, overwise it's a nightmare to use

**Review by atomiix — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thank you @NeOMakinG

**Comment by eternoendless** [comment]
Backported in #1159

**Pull request merged**